Add public library and buffer utility

natebosch opened the pull request:
- Add `buffer` utility and tests
- Export buffer from package library
- Add travis config

natebosch committed:
Add public library and buffer utility

- Add `buffer` utility and tests
- Export buffer from package library
- Add travis config

natebosch commented:
Also see earlier discussion at https://github.com/dart-lang/async/pull/29

natebosch committed:
Add README, CHANGELOG, and version

natebosch committed:
Start with broad travis tasks

jakemac53 commented on lib/src/buffer.dart:
```diff
@@ -0,0 +1,144 @@
+
+    emit() {
+      controller.add(currentResults);
+      currentResults = null;
```
For safety I would just set `waitingForTrigger = true` here, even though it might already be in that state. I think you always want it to be true after this?

natebosch replied:
Good suggestion. Done

kevmoo commented on CHANGELOG.md:
```diff
@@ -0,0 +1,4 @@
+## 0.0.1-dev
```
Don't put `-dev` in the changelog – just in the pubspec.

kevmoo reviewed (approved):
LGTM from a structural perspective.

jakemac53 commented on test/buffer_test.dart:
```diff
@@ -0,0 +1,144 @@
+            [3, 4]
+          ]);
+        });
+
```
A couple other suggestions for tests:

- mixing pre-emptive triggers and normal triggers
- multiple triggers in a row in between values

natebosch replied:
Added 3 new tests around pre-emptive trigger behavior

jakemac53 reviewed (commented):
Couple quick suggestions for some additional tests, but otherwise lgtm.

natebosch committed:
PR suggestions

- Always set waitingForTrigger in emit() - safer that way
- Add more tests around preemptive triggers
- Drop `-dev` from changelog version number
